charms/slurmctld/src/charm.py

```diff
@@ -233,6 +305,21 @@ def _on_slurmd_departed(self, event: SlurmdDepartedEvent) -> None:
             Lack of map between departing unit and NodeName complicates removal of node from gres.conf.
             Instead, rewrite full gres.conf with data from remaining units.
         """
+        # Reconcile the new_nodes.
+        new_nodes = self.new_nodes
+        logger.debug(f"New nodes from stored state: {new_nodes}")
+
+        active_nodes = self._slurmd.get_active_nodes()
+        logger.debug(f"Active nodes: {active_nodes}")
+
+        for node in new_nodes:
+            if node not in active_nodes:
+                logger.debug(f"Removing node: {node}")
+                new_nodes.remove(node)
+
+        # Set the remaining new nodes
+        self.new_nodes = new_nodes
```

Comment:
[question]: Remind me again what prompted adding this bit? Is it due to inconsistent state between what's reported by the relation databags and what's cached by the `new_nodes` property.

Reply:
https://github.com/charmed-hpc/slurm-charms/issues/56

Reply:
When a node goes away, it is currently still accounted for in new_nodes. We need to make sure that new_nodes only contain nodes that still exist.